### PR TITLE
Support to rename project from project manager

### LINKED ISSUE #10740: Add renaming on project select screen
It would be nice to have a button on the project select screen to be able to rename projects.

### PR DESCRIPTION
**Pull request by nabor**
Fixes #10740

### CODE REVIEW

File: editor/project_manager.cpp [hunk 1]
```diff
@@ -443,6 +443,160 @@ class NewProjectDialog : public ConfirmationDialog {
 	}
 };
 
+class RenameProjectDialog : public ConfirmationDialog {
```
Comment (Noshyaar) [inline editor/project_manager.cpp]: Shouldn't this be in project_manager.h?

File: editor/project_manager.cpp [hunk 2]
```diff
@@ -443,6 +443,160 @@ class NewProjectDialog : public ConfirmationDialog {
...
+	RenameProjectDialog() {
+
+		VBoxContainer *vb = memnew(VBoxContainer);
+		add_child(vb);
+
+		Label *l = memnew(Label);
+		l->set_text(TTR("Project Path:"));
+		vb->add_child(l);
+
+
```
Comment (Noshyaar) [inline editor/project_manager.cpp]: Double newlines here. Clang-format won't like that. :P

Comment (nabor) [reply editor/project_manager.cpp]: Ups.

**Comment by nabor** [comment]
Please delete this
I made a new PR with all the changes to avoid polluting the commit history

https://github.com/godotengine/godot/pull/10962

**Pull request closed without merging**